### DISCUSSION TITLE
NTLM auth support in isahc

### DISCUSSION THREAD

**Discussion by run-stop** [sagebind/isahc#329] [Ideas]
I've made a small patch to the src/auth.rs (in my local copy of the isahc) that enables the NTLM support through the CURL library. It works only with dynamically linked CURL, as statically linked CURL for some reason disables the NTLM support. The code at the client side looks as follows:
```
let client = HttpClient::builder()
    .timeout(Duration::from_secs(3))
    .redirect_policy(RedirectPolicy::Follow)
    .authentication(Authentication::ntlm())   
    .credentials(Credentials::new("user","pass"))
    .build()?;
    
let mut response = client.get(url)?;
     println!("{}", response.text()?); 

```
To enable the dynamically linked CURL, you have to use custom features (avoiding a default static CURL). 
In Cargo.toml:

```isahc = {version="1.4.0", default-features=false, features = ["psl", "spnego", "text-decoding"]}```

Please let me know if you find it useful, and I will submit the patch.

**Comment by sagebind** [comment]
See <https://github.com/alexcrichton/curl-rust/issues/359> for more info on why NLTM doesn't work with statically linked curl. Ideally it would be fixed upstream there.
